src/modules/previewpane/MarkdownPreviewHandler/MarkdownPreviewHandlerControl.cs

```diff
@@ -73,28 +70,29 @@ public MarkdownPreviewHandlerControl()
         /// <param name="dataSource">Path to the file.</param>
         public override void DoPreview<T>(T dataSource)
         {
-            this.InvokeOnControlThread(() =>
+            if (!(dataSource is string filePath))

```

Comment:
This is the other change I made, just to throw an exception if the data source isn't a string.  This should never happen, but if something goes wrong this is better than the ArgumentNullException that'll occur in `File.ReadAllText(filePath);`, but I can revert this change in necessary.

I'm also not sure if there's a standard pattern for exceptions like this.

Reply:
@veleek Can we move this logic inside `try` `catch` block in case if it's throws the preview will show the error message(logic in `catch` block) instead of a blank screen.

Reply:
Done